### DISCUSSION TITLE
Should IDE1006 Be Disabled?

### DISCUSSION THREAD

**Discussion by robloo** [AvaloniaUI/Avalonia#9608] [General]
IDE1006 shows the following for `public static readonly` fields with immutable structs which is actually a common use case. The naming rule is inaccurate here and I think it should be disabled altogether. Usage of `s_` is rare anyway and should be for internal/private fields only (although I would argue we shouldn't require this rule there either).

![image](https://user-images.githubusercontent.com/17993847/205450462-6b790a02-6d8b-4093-ac1c-37ebe24dd304.png)

Thoughts?

**Comment by kekekeks** [comment]
Last time we've discussed it back in 2015 we've decided that we should follow .NET naming guidelines rather than ReSharper ones.

**Reply by robloo** [reply]
Makes sense. Since ReSharper and Rider are so widely used with Avalonia projects we might want to adjust the rule settings in the future for these types of things.

**Reply by kekekeks** [reply]
`s_` is the .NET codestyle. ReSharper's default is to use PascalCase for static readonly fields.

**Reply by kekekeks** [reply]
We've adjusted the settings here - https://github.com/AvaloniaUI/Avalonia/blob/c22464eb7d0cb08c4754687a8d031faceec90885/Avalonia.sln.DotSettings#L34-L35

**Reply by robloo** [reply]
Maybe this is a bug in the preview version of VS then. These are public fields, there is no way the rule should apply here.

Personally, I think the modern convention is always to remove the s_ prefix even if this was internal. Private can be developer discretion based on usage IMO.

**Reply by kekekeks** [reply]
@grokys

**Reply by maxkatz6** [reply]
"[.sln.DotSettings](https://github.com/AvaloniaUI/Avalonia/blob/c22464eb7d0cb08c4754687a8d031faceec90885/Avalonia.sln.DotSettings#L34-L35)" is useless without Resharper. We need to move these rules to the editorconfig or VS rulesets.

**Reply by MrJul** [reply]
#10224 disabled the `s_` prefix for public/internal static fields in `.editorconfig`, only private ones are left.

Private static fields through the codebase are currently a mix of `s_`-prefixed and non-prefixed. Even for private fields, when the `s_` is enforced [there are cases where it causes weird naming](https://github.com/AvaloniaUI/Avalonia/pull/10307#discussion_r1103646324).

Should the `s_` rule be removed completely from the `.editorconfig` file? Selectively disabled for some files? Always enforced?

Personally, this discussion and the fact that there are still so many private static fields without `s_` makes me think it the rule should be removed. Whatever the decision is, as an external contributor, I simply want to submit changes that comply with the expected style.

**Reply by robloo** [reply]
Personally, I think it should be disabled completely. Underscores seem to be fine (and public static should have PascalCase in most instances). When the variable is used privately the dev knows if it's static or not. Always seems unnecessary to use this convention -- and a little strange.

**Comment by maxkatz6** [comment]
@robloo it was recently configured to be applied on non-public members only https://github.com/AvaloniaUI/Avalonia/pull/10224
